React/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -1,19 +1,45 @@
 import { stringArrayToObject } from '../../utils/stringUtils';
+import { login as loginService } from '../../services/LoginService';
 
 export const actions = stringArrayToObject(
   [
-    'GET_LOGGED_USER',
-    'LOGIN_SUCCESS'
+    'LOGIN_REQUEST',
+    'LOGIN_SUCCESS',
+    'LOGIN_FAILURE'
   ],
   '@@LOGIN'
 );
 
+const privateActionCreators = {
+  checkResponseValidity: (user, data) => dispatch => {
+    if (data !== undefined && data.password === user.password && data.email === user.email) {
+      dispatch({
+        type: actions.LOGIN_SUCCESS
+      });
+    } else {
+      dispatch(privateActionCreators.getLoginFailure(data));
+    }
+  },
+  getLoginFailure: data => dispatch => {
+    dispatch({
+      type: actions.LOGIN_FAILURE,
+      payload: data === undefined ? 'Email not registered yet' : 'Password is incorrect'
+    });
+  }
+};
+
 const actionCreators = {
-  getLoggedUser: () => ({
-    type: actions.GET_LOGGED_USER
-  }),
-  login: status => dispatch => {
-    dispatch({ type: actions.LOGIN_SUCCESS, payload: status });
+  login: user => async dispatch => {
+    dispatch({ type: actions.LOGIN_REQUEST });
+    const response = await loginService({ email: user.email });
```

Comment:
this is OK, but it will be more difficult to add recompose. So, lets make this: 
1. Let's make loginService to response with ok only when the user has correct pass and email (and you can add a data field in the response with the data you need). In another case, the response should not be ok, and has to have the error, for example in a data field inside the response.. So move this logic: 
` if (data !== undefined && data.password === user.password && data.email === user.email) {`
to the service. 
Then you will have just 2 types of possible answer: success or failure.
2. Change privateActionCreators to loginSuccess and loginFailure.